neqo-transport/tests/retry.rs

```diff
@@ -31,7 +31,7 @@ use test_fixture::{
 
 #[test]
 fn retry_basic() {
-    let server = default_server();
+    let mut server = default_server();
```

Comment:
`Server` is now the sole owner of its `Server::connections` collection. Just a plain `Vec`, no more interior mutability via `Rc<RefCell<_>>`. Thus methods of `Server` need mutable access to `self` in order to mutate `Server::connections`, i.e. in order to process the `Connection`s.